pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -174,6 +181,10 @@ func newRepositoryWithClient(
 	if err != nil {
 		context.GetLogger(ctx).Error(err)
 	}
+	pullthroughinsecure, err := getBoolOption(PullthroughInsecureEnvVar, "pullthroughinsecure", false, options)
+	if err != nil {
+		context.GetLogger(ctx).Error(err)
+	}
```

Comment:
This setting should really be per image stream.  I'm fairly uncomfortable allowing MitM as a global setting.  @liggitt your opinion.